### PR TITLE
Do not replace inside markdown links

### PR DESCRIPTION
**Pull request by andy-held**
`addLinks` replaces terms inside already existing markdown links. This PR changes this behavior such, that markdown links are matched by regex and just replaced with itself in `replace` without change.

Further, a test is introduced that shows the wrong behavior and that the fix actually works. The first two commits simplify the regex replacement by:

- matching the term without a group and removing the second replacement
- employing lookahead assertions

### CODE REVIEW

File: src/test/js/glossary_test.spec.js [hunk 1]
```diff
@@ -109,6 +109,12 @@ describe('Glossary terminology injection', () => {
         expect(result).toContain('## Information about the [ API](/_glossary?id=api)');
     });
 
+    it('preserves case of linked text', () => {
```
Comment (stijn-dejongh) [inline src/test/js/glossary_test.spec.js:112]: Awesome! Good catch

File: src/test/js/glossary_test.spec.js [hunk 2]
```diff
@@ -183,6 +189,19 @@ describe('Glossary terminology injection', () => {
 
         expect([...result.matchAll('/_glossary\\?id=api')]).toHaveLength(0);
     });
+
+    it('Word replacements do not replace parts of a link', () => {
```
Comment (stijn-dejongh) [inline src/test/js/glossary_test.spec.js:193]: I completely missed this edge case. Thank you for fixing it

**Review by stijn-dejongh — APPROVED** [review]
Thank you for your work. I like your changes to the RegExp, as they make for a less convoluted step-by-step process. I appreciate the effort in writing those test cases to illustrate the issue and avoid regressions.

**Comment by stijn-dejongh** [comment]
I fixed one test case that was missed during your code changes.
Merging this PR in. Thank you for contributing!
